pkg/workflows/wasm/host/module.go

```diff
@@ -143,7 +150,7 @@ func WithDeterminism() func(*ModuleConfig) {
 	}
 }
 
-func NewModule(modCfg *ModuleConfig, binary []byte, opts ...func(*ModuleConfig)) (*module, error) {
+func NewModule(ctx context.Context, modCfg *ModuleConfig, binary []byte, opts ...func(*ModuleConfig)) (*module, error) {
```

Comment:
I tried to find a way to avoid breaking this signature, but there's not an elegant way around it without just introducing a new func to use instead. Happy to do that instead if folks prefer 🤷

Reply:
I suppose another variation would be to extract the binary size checks to be pre-construction in order to avoid the issue and simplify this implementation. The fact that we already check the binary size elsewhere suggests that we could simplify a bit too.